src/git/prod_repo.go

```diff
@@ -26,3 +26,21 @@ func NewProdRepo() *ProdRepo {
 		Configuration: config,
 	}
 }
+
+// RemoveOutdatedConfiguration removes outdated Git Town configuration.
+func (r *ProdRepo) RemoveOutdatedConfiguration() error {
+	for child, parent := range r.GetParentBranchMap() {
+		hasChildBranch, err := r.Silent.HasLocalOrRemoteBranch(child)
+		if err != nil {
+			return err
+		}
+		hasParentBranch, err := r.Silent.HasLocalOrRemoteBranch(parent)
+		if err != nil {
+			return err
+		}
+		if !hasChildBranch || !hasParentBranch {
+			r.Configuration.DeleteParentBranch(child)
```

Comment:
Interesting. Just thinking on this more how Configuration has its own shell. Any thoughts if there are other "libraries" like within a gitRepo we would want to break out to function similar to Configuration? Would probably need to be things that always run silently.

Conversely, thoughts on removing Configuration as a special case?

Reply:
Yes, the new architecture consists of packages that are orthogonal to each other and fully independent of each other, i.e. no dependencies between them. In the ideal case that I try to reach, the `git` package defines a `git.Shell` interface that describes the shell methods that the `git` package needs to execute. It is then given an instance of the `command.SilentShell` struct that implements this interface. This means there are no dependencies between the `git` and the `command` packages but they compose well and integrate each other deeply. 

There might be additional aspects that could be extracted into such independent, orthogonal, composable packages. My current list is [here](https://github.com/git-town/git-town/blob/master/documentation/development/architecture.md). I think you are right that the configuration is independent from running Git commands. The fact that we store configuration in Git is just an implementation detail. There should probably be a `config` package.

It's interesting that the Go syntax can be learned in a week, but it took us (or at least me) years to understand how to use it the right way. 😃 It's a bit like the game Go in this sense. Or maybe Cyvasse!